### PR TITLE
Dynamic Simulation - Curve configuration for BUS

### PR DESCRIPTION
**Pull request by thangqp**
Take into account equipmentType (as static or dynamic type) of curve info while generating the script groovy

_**Related PRs**_

**Front-end :** 
https://github.com/gridsuite/gridstudy-app/pull/1763 (with demo)

**Back-end :**
https://github.com/gridsuite/filter-server/pull/82
https://github.com/gridsuite/study-server/pull/489

### CODE REVIEW

File: src/main/java/org/gridsuite/ds/server/utils/EquipmentType.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package org.gridsuite.ds.server.utils;
```
Comment (EstherDarkish) [inline src/main/java/org/gridsuite/ds/server/utils/EquipmentType.java:1]:
```suggestion
/**
 * Copyright (c) 2024, RTE (http://www.rte-france.com)
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */
 package org.gridsuite.ds.server.utils;
```

**Review by EstherDarkish — APPROVED** [review]

File: src/main/java/org/gridsuite/ds/server/utils/EquipmentType.java (hunk 1 above)
Comment (thangqp) [reply src/main/java/org/gridsuite/ds/server/utils/EquipmentType.java:7]: DONE by https://github.com/gridsuite/dynamic-simulation-server/pull/79/commits/d4f4d20e39fe53c61ea028c75d1068d93d0cb96a

File: src/main/java/org/gridsuite/ds/server/service/parameters/impl/CurveGroovyGeneratorServiceImpl.java [hunk 2]
```diff
@@ -39,13 +42,20 @@ public String generate(List<CurveInfos> curveInfosList) {
...
+        // group variables by equipmentId
         Map<String, String> variablesByEquipmentIdMap = curveInfosList.stream()
-                .collect(Collectors.groupingBy(CurveInfos::getEquipmentId, Collectors.mapping(curveInfo -> "\"" + curveInfo.getVariableId() + "\"", Collectors.joining(", "))));
+                .collect(Collectors.groupingBy(CurveInfos::getEquipmentId, LinkedHashMap::new,
+                        Collectors.mapping(curveInfo -> "\"" + curveInfo.getVariableId() + "\"", Collectors.joining(", "))));
 
         String[] curveStringList = variablesByEquipmentIdMap.entrySet().stream().map(variablesByEquipmentId -> {
             ST curveST = new ST(curveTemplate);
-            curveST.add("dynamicModelId", "\"" + variablesByEquipmentId.getKey() + "\"");
+            curveST.add("idName", isStaticType(equipmentTypeByEquipmentIdMap.get(variablesByEquipmentId.getKey())) ? "staticId" : "dynamicModelId");
```
Comment (flomillot) [inline src/main/java/org/gridsuite/ds/server/service/parameters/impl/CurveGroovyGeneratorServiceImpl.java:57]:
```suggestion
            curveST.add("idType", isStaticType(equipmentTypeByEquipmentIdMap.get(variablesByEquipmentId.getKey())) ? "staticId" : "dynamicModelId");
```
Or I misunderstood ? Why `idName` ?

Comment (thangqp) [reply src/main/java/org/gridsuite/ds/server/service/parameters/impl/CurveGroovyGeneratorServiceImpl.java:57]: In fact, i am generating a pair 'key value' for field 'id' in the template file, https://github.com/gridsuite/dynamic-simulation-server/pull/79/files#diff-2c7ff1d4283efb35791e70498385c45ea5e5e3dcf6bf41ae87064f098020217aR2

May be 'idKey' is better than 'idName' and 'idType'

Comment (thangqp) [reply src/main/java/org/gridsuite/ds/server/service/parameters/impl/CurveGroovyGeneratorServiceImpl.java]: DONE by https://github.com/gridsuite/dynamic-simulation-server/pull/79/commits/4ca8d7423180964321314354ccdabf4627f3f363

**Review by flomillot — APPROVED** [review]

**Pull request merged**